remote/remote.go

```diff
@@ -194,11 +194,13 @@ func newV1Image(keychain authn.Keychain, repoName string, platform imgutil.Platf
 		image, err = remote.Image(ref, remote.WithAuth(auth), remote.WithTransport(http.DefaultTransport), remote.WithPlatform(v1Platform))
 		if err != nil {
 			if err == io.EOF && i != maxRetries {
-				continue // retry if EOF
+				continue // retry
 			}
-			if transportErr, ok := err.(*transport.Error); ok && len(transportErr.Errors) > 0 {
-				switch transportErr.StatusCode {
-				case http.StatusNotFound, http.StatusUnauthorized:
+			if transportErr, ok := err.(*transport.Error); ok && is40x(transportErr) {
+				if i != maxRetries {
+					continue // retry
+				}
+				if len(transportErr.Errors) > 0 {
```

Comment:
I am not familiar with the `transport.Error` - what is the purpose of the guard `len(transportErr.Errors) > 0`? I see it existed before, but was it guarding a failure of `transportErr.StatusCode`? The history of the files makes it look like we used to have this guard to protect `transportErr.Errors[0].Code` - but since we aren't doing that anymore I think we could remove this check entirely.